### PR TITLE
Feature/Google signIn using additional scopes

### PR DESCRIPTION
**Pull request by EgzonArifi**
Support additional scopes for google sign in

### CODE REVIEW

File: Sources/Google/GoogleAuthenticator.swift [hunk 1]
```diff
@@ -24,7 +24,9 @@ extension GoogleAuthenticator: Authenticator {
   /// SignIn user.
   ///
   /// Will return promise with the `Response` object on success or with `Error` on error.
-  public func signIn(from presentingViewController: UIViewController) -> Promise<Response> {
+  public func signIn(from presentingViewController: UIViewController,
+                     hint: String? = .none,
+                     scopes: [String]? = .none) -> Promise<Response> {
```
Comment (borut-t) [inline Sources/Google/GoogleAuthenticator.swift:29]: What is a `hint` and what is `scopes`?

Comment (EgzonArifi) [reply Sources/Google/GoogleAuthenticator.swift:29]: **hint**
An optional hint for the authorization server, for example the user’s ID or email address, to be prefilled if possible.

**additionalScopes**
An optional array of scopes to request in addition to the basic profile scopes.

Comment (EgzonArifi) [reply Sources/Google/GoogleAuthenticator.swift]: when using GmailAPI for example, we need to request additional authentication scopes to manage user emails like fetching, reply etc.

**Review by borut-t — APPROVED** [review]
Great 👌

**Pull request merged**